godot-core/src/builtin/plane.rs

```diff
@@ -105,16 +107,134 @@ impl Plane {
+	    dist <= tolerance.unwrap_or(CMP_EPSILON)
+    }
+
+    /// Finds the intersection point of three planes.
+    ///
+    /// Returns the intersection point of the `Plane`s of the current `Plane`, `b`, and `c` enclosed
+    /// in `Some`. If no intersection point is found, `None` will be returned.
+    #[inline]
+    pub fn intersect_3(&self, b: &Self, c: &Self) -> Option<Vector3> {
+        let normal0: Vector3 = self.normal;
+	    let normal1: Vector3 = b.normal;
+	    let normal2: Vector3 = c.normal;
+        let denom: real = normal0.cross(normal1).dot(normal2);
+        if is_zero_approx(denom) {
+            return None;
+        }
+        let result = normal1.cross(normal2) * self.d + normal2.cross(normal0) * b.d + normal0.cross(normal1) * c.d;
+        Some(result / denom)
+    }
+
+    /// Finds the intersection point of the plane with a ray.
+    ///
+    /// Returns the intersection point of a ray consisting of the position `from` and the direction 
+    /// normal `dir` with the `self` `Plane` enclosed in `Some`. If no intersection is found, `None`
+    /// will be returned.
+    #[inline]
+    pub fn intersects_ray(&self, from: Vector3, dir: Vector3) -> Option<Vector3> {
+	    let denom: real = self.normal.dot(dir);
+        if is_zero_approx(denom) {
+            return None;
+        }
+        let dist: real = (self.normal.dot(from) - self.d) / denom;
+        if dist > CMP_EPSILON {
+            return None;
+        }
+        Some(from - dir * dist)
+    }
+
+    /// Finds the intersection point of the plane with a line segment.
+    /// 
+    /// Returns the intersection point of a segment from position `from` to position `to` with the 
+    /// `self` `Plane` enclosed in `Some`. If no intersection is found, `None` will be returned.
+    #[inline]
+    pub fn intersects_segment(&self, from: Vector3, to: Vector3) -> Option<Vector3> {
+        let segment: Vector3 = from - to;
+        let denom: real = self.normal.dot(segment);
+        if is_zero_approx(denom) {
+            return None;
+        }
+        let dist: real = (self.normal.dot(from) - self.d) / denom;
+        if dist < -CMP_EPSILON || dist > (1.0 + CMP_EPSILON) {
+            return None;
+        }
+        Some(from - segment * dist)
+    }
+
+    /// Finds whether the two planes are approximately equal.
+    ///
     /// Returns `true` if the two `Plane`s are approximately equal, by calling `is_equal_approx` on
     /// `normal` and `d` or on `-normal` and `-d`.
-    ///
-    /// _Godot equivalent: `Plane.is_equal_approx()`_
     #[inline]
     pub fn is_equal_approx(&self, other: &Self) -> bool {
         (self.normal.is_equal_approx(other.normal) && is_equal_approx(self.d, other.d))
             || (self.normal.is_equal_approx(-other.normal) && is_equal_approx(self.d, -other.d))
     }
 
+    /// Finds whether the plane is finite or not.
+    ///
+    /// Returns `true` if the `self` `Plane` is finite by calling `is_finite` on `normal` and `d`. 
```

Comment:
The first sentence here seems a bit redundant to me. `is_finite` already communicates that this function finds whether the plane is finite or not. I think you can just cut out the first sentence. Same for `is_point_over`.